stream-chat-android-state/src/main/java/io/getstream/chat/android/state/plugin/logic/channel/thread/internal/QueryThreadsStateLogic.kt

```diff
@@ -0,0 +1,162 @@
+ *
+ * @param mutableState Reference to the global [QueryThreadsMutableState].
+ */
+internal class QueryThreadsStateLogic(private val mutableState: QueryThreadsMutableState) {
+
+    /**
+     * Retrieves the current state of the 'loading' indicator from the [mutableState].
+     */
+    internal fun isLoading() = mutableState.loading.value
+
+    /**
+     * Updates the loading state of the [mutableState].
+     *
+     * @param loading The new loading state.
+     */
+    internal fun setLoading(loading: Boolean) =
+        mutableState.setLoading(loading)
+
+    /**
+     * Retrieves the current state of the 'loading more' indicator from the [mutableState].
+     */
+    internal fun isLoadingMore() = mutableState.loadingMore.value
+
+    /**
+     * Updates the loading more state of the [mutableState].
+     *
+     * @param loading The new loading more state.
+     */
+    internal fun setLoadingMore(loading: Boolean) =
+        mutableState.setLoadingMore(loading)
+
+    /**
+     * Retrieves the current state of the thread list from the [mutableState].
+     */
+    internal fun getThreads() = mutableState.threads.value
+
+    /**
+     * Updates the thread state of the [mutableState].
+     *
+     * @param threads The new threads state.
+     */
+    internal fun setThreads(threads: List<Thread>) =
+        mutableState.setThreads(threads)
+
+    /**
+     * Appends the new page of [threads] to the current thread list.
+     *
+     * @param threads The new page of threads.
+     */
+    internal fun appendThreads(threads: List<Thread>) =
+        mutableState.appendThreads(threads)
+
+    /**
+     * Updates the identifier for the next page of threads in the [mutableState].
+     *
+     * @param next The next page identifier.
+     */
+    internal fun setNext(next: String?) =
+        mutableState.setNext(next)
+
+    /**
+     * Adds a new thread to the set of unseen thread IDs in the [mutableState].
+     *
+     * @param id The ID of the new [Thread].
+     */
+    internal fun addUnseenThreadId(id: String) =
+        mutableState.addUnseenThreadId(id)
+
+    /**
+     * Clears the set of unseen thread IDs in the [mutableState].
+     */
+    internal fun clearUnseenThreadIds() =
+        mutableState.clearUnseenThreadIds()
+
+    /**
+     * Updates the parent message of a thread.
+     *
+     * @param parent The new state of the thread parent message.
+     * @return true if matching parent message was found and was updated, false otherwise.
+     */
+    internal fun updateParent(parent: Message): Boolean {
+        val oldThreads = getThreads()
+        val oldThread = oldThreads.find { it.parentMessageId == parent.id }
+        oldThread ?: return false // no matching parent message was found
+        val newThread = oldThread.copy(
+            parentMessage = parent,
+            deletedAt = parent.deletedAt,
+            updatedAt = parent.updatedAt,
+            replyCount = parent.replyCount,
```

Comment:
I believe that this would work for `deletedAt` and `replyCount`, but I am not sure if we can fully rely on the `parentMessage.updatedAt` to calculate the thread's `updatedAt` value. (not sure at this point if a thread can be updated by some other means, other than updating its parent message)